engines/vcruise/vcruise.cpp

```diff
@@ -0,0 +1,187 @@
+
+	const Graphics::PixelFormat *fmt16_565 = nullptr;
+	const Graphics::PixelFormat *fmt16_555 = nullptr;
+	const Graphics::PixelFormat *fmt32 = nullptr;
+
+	for (const Graphics::PixelFormat &fmt : pixelFormats) {
+		if (fmt32 == nullptr && fmt.bytesPerPixel == 4 && fmt.rBits() == 8 && fmt.gBits() == 8 && fmt.bBits() == 8)
+			fmt32 = &fmt;
+		if (fmt16_555 == nullptr && fmt.rBits() == 5 && fmt.gBits() == 5 && fmt.bBits() == 5)
+			fmt16_555 = &fmt;
+		if (fmt16_565 == nullptr && fmt.rBits() == 5 && fmt.gBits() == 6 && fmt.bBits() == 5)
+			fmt16_565 = &fmt;
+	}
+
+	// Figure out screen layout
+	Common::Point size;
+
+	Common::Point videoSize;
+	Common::Point traySize;
+	Common::Point menuBarSize;
+	const char *exeName = nullptr;
+
+	if (_gameDescription->gameID == GID_REAH) {
+		videoSize = Common::Point(608, 348);
+		menuBarSize = Common::Point(640, 44);
+		traySize = Common::Point(640, 88);
+		exeName = "Reah.exe";
+	} else if (_gameDescription->gameID == GID_SCHIZM) {
+		videoSize = Common::Point(640, 360);
+		menuBarSize = Common::Point(640, 32);
+		traySize = Common::Point(640, 88);
+		exeName = "Schizm.exe";
+	} else {
+		error("Unknown game");
+	}
+
+	size.x = videoSize.x;
+	if (menuBarSize.x > size.x)
+		size.x = menuBarSize.x;
+	if (traySize.x > size.x)
+		size.x = traySize.x;
+
+	size.y = videoSize.y + menuBarSize.y + traySize.y;
+
+	Common::Point menuTL = Common::Point((size.x - menuBarSize.x) / 2, 0);
+	Common::Point videoTL = Common::Point((size.x - videoSize.x) / 2, menuTL.y + menuBarSize.y);
+	Common::Point trayTL = Common::Point((size.x - traySize.x) / 2, videoTL.y + videoSize.y);
+
+	_menuBarRect = Common::Rect(menuTL.x, menuTL.y, menuTL.x + menuBarSize.x, menuTL.y + menuBarSize.y);
+	_videoRect = Common::Rect(videoTL.x, videoTL.y, videoTL.x + videoSize.x, videoTL.y + videoSize.y);
+	_trayRect = Common::Rect(trayTL.x, trayTL.y, trayTL.x + traySize.x, trayTL.y + traySize.y);
+
+	if (fmt32)
+		initGraphics(size.x, size.y, fmt32);
+	else if (fmt16_565)
+		initGraphics(size.x, size.y, fmt16_565);
+	else if (fmt16_555)
+		initGraphics(size.x, size.y, fmt16_555);
+	else
+		error("Unable to find a suitable graphics format");
+
+	_system->fillScreen(0);
+
+	_runtime.reset(new Runtime(_system, _mixer, _rootFSNode, _gameDescription->gameID));
+	_runtime->initSections(_videoRect, _menuBarRect, _trayRect, _system->getScreenFormat());
+
+	_runtime->loadCursors(exeName);
+
+	if (ConfMan.getBool("vcruise_debug")) {
+		_runtime->setDebugMode(true);
+	}
+
+	// Run the game
+	while (!shouldQuit()) {
+		handleEvents();
+
+		if (!_runtime->runFrame())
+			break;
+
+		_runtime->drawFrame();
+		_system->delayMillis(10);
+	}
+
+	_runtime.reset();
+
+	return Common::kNoError;
+}
+
+void VCruiseEngine::pauseEngineIntern(bool pause) {
+	Engine::pauseEngineIntern(pause);
+}
+
+bool VCruiseEngine::hasFeature(EngineFeature f) const {
+	switch (f) {
+	case kSupportsReturnToLauncher:
+	case kSupportsSavingDuringRuntime:
```

Comment:
I see no save support yet. This needs to be disabled for the time being so the GMM does not get confused.